src/kotlin_editor_export_plugin.cpp

```diff
@@ -40,21 +40,55 @@ void KotlinEditorExportPlugin::_export_begin(const HashSet<String>& p_features,
                 bool is_x64 {p_features.has("x86_64")};
 
                 if (!is_arm64 && !is_x64) {
-                    add_macos_plugin_file(String(RES_DIRECTORY).path_join(EMBEDDED_JRE_DIRECTORY));
+                    add_macos_plugin_file(String(RES_DIRECTORY).path_join(HOST_EMBEDDED_JRE_DIRECTORY));
                 } else {
-                    if (is_arm64) { add_macos_plugin_file(String(RES_DIRECTORY).path_join(EMBEDDED_JRE_ARM_DIRECTORY)); }
-                    if (is_x64) { add_macos_plugin_file(String(RES_DIRECTORY).path_join(EMBEDDED_JRE_AMD_DIRECTORY)); }
+                    if (is_arm64) { add_macos_plugin_file(String(RES_DIRECTORY).path_join(MACOS_EMBEDDED_JRE_ARM_DIRECTORY)); }
+                    if (is_x64) { add_macos_plugin_file(String(RES_DIRECTORY).path_join(MACOS_EMBEDDED_JRE_AMD_DIRECTORY)); }
                 }
             } else {
```

Comment:
We talked about it last time. We don't support all system and architecture that godot does, so we can't assume that the generic "else" works in all cases. We should make the if extra explicit and only leave a final else to raise an error, "Platform not supported"). We can also simplify its structure a bit compared to what it is now.

Reply:
replaced the generic else with a proper check for universal for macos and failure in all other cases
restructured the export a bit. it should now represent what we've discussed last time IIRC